sklearn/metrics/tests/test_pairwise.py

```diff
@@ -407,6 +417,72 @@ def test_euclidean_distances():
     assert_greater(np.max(np.abs(wrong_D - D1)), .01)
 
 
+def test_masked_euclidean_distances():
+    # Check with pairs of matrices with missing values
+    X = np.array([[1., np.nan, 3., 4., 2.],
+                  [np.nan, 4., 6., 1., np.nan],
+                  [3., np.nan, np.nan, np.nan, 1.]])
+
+    Y = np.array([[np.nan, 7., 7., np.nan, 2.],
+                  [np.nan, np.nan, 5., 4., 7.],
+                  [np.nan, np.nan, np.nan, 4., 5.]])
+
+    D1 = masked_euclidean_distances(X, Y,  missing_values="NaN")
+    D2 = masked_euclidean_distances(X, Y, squared=True, missing_values="NaN")
+
+    assert_array_almost_equal(D1**2, D2)
+
+    # Check when squared = True
+    D3 = np.array(
+        [[40., 48.33333331, 22.5],
+         [25., 25., 45.],
+         [5., 180., 80.]])
+    D4 = masked_euclidean_distances(X, Y, squared=True, missing_values="NaN")
+
+    assert_array_almost_equal(D3, D4)
```

Comment:
Isn't D4 the same as D2?

Reply:
Sure, but I think I used D4 because it is a separate test from the D2 test so was trying to make them "independent". But I will change this if you want?

Reply:
I would reduce the complexity/amount of the code if possible